Domains: Update pricing and description text in "Use a domain I own" flow

leonardost opened the pull request:
#### Changes proposed in this Pull Request

When users select the "Use a domain I own" flow and input a domain they want to use, they are presented with some inaccurate information regarding the plans which include each upgrade:

![Markup on 2022-03-31 at 18:17:50](https://user-images.githubusercontent.com/5324818/161152733-870fcef9-98b8-4bd8-8734-54c411397261.png)

Notice the text says "Included with annual plans", which should be replaced with the new Pro plan. Also, the standalone pricing for the domain connection is shown but it shouldn't, as it can only be added if the user has a plan.

This is how this page looks with this PR:

![Markup on 2022-03-31 at 18:15:55](https://user-images.githubusercontent.com/5324818/161153033-d6c64f8c-a79c-4a8e-9487-32dd78859094.png)

#### Testing instructions

- Build this branch locally or open the live Calypso link
- Access `/start/domains`
- Select the "Use a domain I own" flow
- Input a domain
- Ensure the transfer or connection page looks like the screenshot above

leonardost committed:
Update pricing and description text in "Use a domain I own" flow

delputnam reviewed (approved):
This LGTM.  Work as expected.  Thank you! 🙇

leonardost committed:
Hide maping price in transfer or connect selection page

We should not show the standalone mapping price anymore because all
users need a plan to connect domains. Even PWPO-exempt users need to
buy a plan if they want to buy new mappings.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7219378

Thank you @leonardost for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7219378).
